QModManager/Utility/IOUtilities.cs

```diff
@@ -22,31 +22,36 @@ internal static class IOUtilities
             "steam_shader_cache",
         };
 
-        internal static string GetFolderStructureAsTree(string directory = null)
+        internal static void LogFolderStructureAsTree(string directory = null)
         {
             try
             {
                 directory ??= Environment.CurrentDirectory;
 
-                return GenerateFolderStructure(directory);
+                GenerateFolderStructure(directory);
             }
             catch (Exception e)
             {
                 throw e;
             }
         }
 
-        internal static string GenerateFolderStructure(string directory)
+        internal static void GenerateFolderStructure(string directory)
         {
             var builder = new StringBuilder();
+
             try
             {
                 builder.AppendLine();
                 builder.AppendLine($"+ {new DirectoryInfo(directory).Name}");
 
+                Logger.Info(builder.ToString());
+                builder.Clear();
                 foreach (string dir in Directory.GetDirectories(directory))
                 {
                     GetFolderStructureRecursively(builder, dir, 0);
+                    Logger.Info(builder.ToString());
```

Comment:
Also: Using `Logger.Info` here will result an additional change in the output: each line of the tree will be prefixed with "`[QModManager:INFO] `" which IMO is just added noise and a waste of bytes. I would try to use `Console.WriteLine` instead, except for the initial string "`Folder structure:`".

Reply:
By convention, we always want to know where a log is coming from.  
Though for this, I can see how it might actually make the log more messy than it has to be.
:+1:

Reply:
> By convention, we always want to know where a log is coming from.
> Though for this, I can see how it might actually make the log more messy than it has to be.
> 👍

While I do agree with that in general, there is a precedent already set in QMM where we are not prefixing each line, but only the first, such as the `SummaryLogger` class. Using `Console.WriteLine()` here for each line *after* the initial has the same effect as `builder.AppendLine()` in the output.

Reply:
> The intent was to take the contents of each, top-level folder and print that as a single log.
> That would still include several line breaks within the log.

Since I misinterpreted your original intent, I'm not sure my comment here is still valid. I would need to see some sample output before I can make a conclusion :)

Reply:
No, I agree that making an exception in this instance, and keeping the final output in the file formatted the same way as it was before, would be the better choice.  
Do do that, we would have to bypass the `Logger` and go straight through `Console.WriteLine` to avoid prepending the extra meta-data on each log.